league_rpc_linux/processes/lcu_thread.py

```diff
@@ -0,0 +1,355 @@
+        + ": "
+        + str(rankedData["highestRankedEntry"]["leaguePoints"])
+        + " LP"
+    )
+
+    lobbyRawData = await connection.request(
+        "GET", "/lol-gameflow/v1/gameflow-metadata/player-status"
+    )
+    lobbyData = await lobbyRawData.json()
+    data["queue_id"] = lobbyData["currentLobbyStatus"]["queueId"]
+    data["lobby_id"] = lobbyData["currentLobbyStatus"]["lobbyId"]
+    data["players"] = len(lobbyData["currentLobbyStatus"]["memberSummonerIds"])
+
+    if data["queue_id"] == -1:
+        # Not in Lobby
+        return
+
+    data["inLobby"] = True
+    lobbyQueueInfoRaw = await connection.request(
+        "GET", "/lol-game-queues/v1/queues/" + str(data["queue_id"])
+    )
+    lobbyQueueInfo = await lobbyQueueInfoRaw.json()
+    data["queue"] = lobbyQueueInfo["name"]
+    data["max_players"] = int(lobbyQueueInfo["maximumParticipantListSize"])
+    data["map_id"] = lobbyQueueInfo["mapId"]
+    data["gamemode"] = lobbyQueueInfo["gameMode"]
+    data["lobbyQueueIsRanked"] = lobbyQueueInfo["isRanked"]
+
+
+###### Debug ######
+
+
+# This will catch all events and print them to the console.
+# @connector.ws.register("/", event_types=("UPDATE", "CREATE", "DELETE"))
+# async def debug(connection, event):
+#     print(f"DEBUG - {event.type}: {event.uri}")
+
+
+def update_rpc():
+    print("Updating Discord Presence.")
+
+    if data["gameflowPhase"] == "InProgress":
+        # In Game - handled by main process
+        return
+
+    if not data["inLobby"]:
+        rpc.update(
+            large_image=f"{PROFILE_ICON_BASE_URL}{str(data['summoner_icon'])}.png",
+            large_text="In Client",
+            small_image=LEAGUE_OF_LEGENDS_LOGO,
+            small_text="League of Legends Logo",
+            details="In Client",
+            state=f"{data['availability']}",
+        )
+        return
+
+    if data["inChampSelect"]:
+        # In Champ Select
+        rpc.update(
+            large_image=BASE_MAP_ICON_URL.format(
+                map_name=MAP_ICON_CONVERT_MAP.get(data["map_id"])
+            ),
+            large_text=f"{GAME_MODE_CONVERT_MAP.get(data['gamemode'], data['gamemode'])}",
+            small_image=LEAGUE_OF_LEGENDS_LOGO,
+            small_text="League of Legends Logo",
+            details=f"In Champ Select: {data['queue']}",
+            state=f"Picking Champions...",
+            party_size=[data["players"], data["max_players"]],
+        )
+        return
+
+    if data["inQueue"]:
+        # In Queue
+        rpc.update(
+            large_image=BASE_MAP_ICON_URL.format(
+                map_name=MAP_ICON_CONVERT_MAP.get(data["map_id"])
+            ),
+            large_text=f"{GAME_MODE_CONVERT_MAP.get(data['gamemode'], data['gamemode'])}",
+            small_image=LEAGUE_OF_LEGENDS_LOGO,
+            small_text="League of Legends Logo",
+            details=f"In Queue: {data['queue']}",
+            state=f"Searching for Game...",
```

Comment:
Sine no variables are used inside, we don't have to turn this string into an "f string"